task_geo/data_sources/covid/fr_covidata/fr_covidata.py

```diff
@@ -0,0 +1,257 @@
+"""
+fr_covidata.py
+
+Functions:
+    - fr_covidata_connector: Extracts data from CSV URL
+    - fr_covidata_formatter: Cleans CSV data
+    - fr_covidata: Combines the two previous functions
+
+Data Credits:
+    OpenCOVID19-fr
+    https://www.data.gouv.fr/en/datasets/chiffres-cles-concernant-lepidemie-de-covid19-en-france/
+    https://github.com/opencovid19-fr/data
+"""
+
+import io
+
+import pandas as pd
+import requests
+
+url = (
+    'https://raw.githubusercontent.com/opencovid19-fr/'
+    'data/master/dist/chiffres-cles.csv'
+)
+
+
+def fr_covidata():
+    """Data Source for the French COVID-19 Data.
+    Arguments:
+        None
+    Returns:
+        pandas.DataFrame
+    """
+    df = fr_covidata_connector()
+    return fr_covidata_formatter(df)
+
+
+def fr_covidata_connector():
+    """Extract data from OpenCOVID19-fr's Github repository.
+    Description:
+        - Downloads the URL's data in a Unicode CSV Format
+        - Unicode CSV Format: ACS 5Y UTF-8
+    Returns:
+        dataset (DataFrame with CSV Data)
+    """
+
+    urlData = requests.get(url).content
+
+    dataset = pd.read_csv(io.StringIO(urlData.decode('utf-8')))
+    return dataset
+
+
+def fr_covidata_formatter(dataset):
+    """Formatter for FR COVID-19 Data.
+    Arguments:
+        dataset(pandas.DataFrame): Data as returned by fr_covidata_connector.
+    Description:
+        - Drop unnecessary rows with irrelevant regions' info and only keep
+        info related to subregions in Metropolitan France, as well as
+        repetitive data
+        - Check the dataset for instances where there are more than one source
+        of data in the same subregion for the same date, then complement all
+        the sources information, and take the highest value in case there are
+        different values for the same column, while aggregating the sources
+        info
+        - Rename/Translate the column titles, and add a country column (France)
+    Returns:
+        frcovidata(pandas.DataFrame)
+    """
+
+    dataset = dataset[dataset.granularite != 'region']
+    dataset = dataset[dataset.granularite != 'monde']
+    dataset = dataset[dataset.granularite != 'pays']
+    dataset = dataset[dataset.granularite != 'collectivite-outremer']
+    dataset = dataset[dataset.maille_code != 'DEP-971']
+    dataset = dataset[dataset.maille_code != 'DEP-972']
+    dataset = dataset[dataset.maille_code != 'DEP-973']
+    dataset = dataset[dataset.maille_code != 'DEP-974']
+    dataset = dataset[dataset.maille_code != 'DEP-976']
```

Comment:
This can be done in 3 lines with:
```python
no_granularites = ['region', 'monde', 'pays', 'collectivite-outremer']
no_maille_codes = ['DEP-971', ....]
dataset = dataset[
    (~dataset.granularite.isin(granularites)) & (~dataset.maille_code.isin(no_maille_codes))
]